### ISSUE TITLE
Add cellsize check to CatalogFFTPower

### ISSUE THREAD

**Issue by seshnadathur** [cosmodesi/pypower#13]
For survey power calculation, using the `nmesh` argument to specify the mesh grid size is not so useful, because one doesn't know how large the box should be/will be (expecially when determining the boxsize automatically from the positions and adding a padding using `boxpad`). It makes more sense then to use `cellsize` to specify the desired resolution of the grid.

However, when I do this I often get an error saying that the resultant mesh size is odd and odd meshes are not supported, and the code then fails. But I can't know in advance whether `boxsize/cellsize` is going to be an odd or even number, since the point of this use case scenario is that I don't know `boxsize`!

I suggest that in such a case the code should simply print a warning, choose the next largest even grid size and continue without error.

**Comment by adematti** [comment]
Good catch, thanks! Should be fixed by [fe98f07](https://github.com/cosmodesi/pypower/commit/fe98f07c642a3259fc475c249150cf742d553afe).
If that seems fine, I will update the cosmodesi environment at NERSC.

**Comment by seshnadathur** [comment]
I looked through the code and it looks fine. Unfortunately I can't test it in my use case until is updated in the cosmodesi environment at NERSC!

**Comment by adematti** [comment]
pypower has been updated in the "main" version of the cosmodesi environment.
